Core.ApplicationServices/SystemUsage/Migration/ItSystemUsageMigrationServiceAdapter.cs

```diff
@@ -0,0 +1,122 @@
+                );
+        }
+
+        public Result<ItSystemUsage, OperationError> ExecuteMigration(Guid usageUuid, Guid toSystemUuid)
+        {
+            var transaction = _transactionManager.Begin();
+
+            return ResolveUsageAndSystemIds(usageUuid, toSystemUuid)
+                .Bind(result => _systemUsageMigrationService.ExecuteSystemUsageMigration(result.usageId, result.systemId))
+                .Bind(WithModifyAccess)
+                .Match(
+                    usage =>
+                    {
+                        transaction.Commit();
+                        return Result<ItSystemUsage, OperationError>.Success(usage);
+                    },
+                    error =>
+                    {
+                        transaction.Rollback();
+                        return error;
+                    }
+                );
+        }
+
+        public Result<IEnumerable<ItSystem>, OperationError> GetUnusedItSystemsByOrganization(Guid organizationUuid,
+            int numberOfItSystems,
+            bool getPublicFromOtherOrganizations,
+            params IDomainQuery<ItSystem>[] conditions)
+        {
+            return _organizationService.GetOrganization(organizationUuid)
+                .Bind(organization => _systemUsageMigrationService.GetUnusedItSystemsByOrganizationQuery(organization.Id, numberOfItSystems, getPublicFromOtherOrganizations, conditions))
+                .Select<IEnumerable<ItSystem>>(x => x.ToList());
+        }
+
+        public IEnumerable<CommandPermissionResult> GetCommandPermissions()
+        {
+            var commandPermissions = new List<CommandPermissionResult>
+            {
+                new (CommandPermissionConstraints.UsageMigration.Execute, _systemUsageMigrationService.CanExecuteMigration())
+            };
+
+            return commandPermissions;
+        }
+
+        private Result<ItSystemUsage, OperationError> WithReadAccess(ItSystemUsage usage)
+        {
+            return _authorizationContext.AllowReads(usage)
+                ? usage
+                : new OperationError($"User is not allowed to read It System Usage with uuid: {usage.Uuid}", OperationFailure.Forbidden);
+        }
+
+        private Result<ItSystemUsage, OperationError> WithModifyAccess(ItSystemUsage usage)
+        {
+            return _authorizationContext.AllowModify(usage)
+                ? usage
+                : new OperationError($"User is not allowed to modify It System Usage with uuid: {usage.Uuid}", OperationFailure.Forbidden);
+        }
+
+        private Result<(int usageId, int systemId), OperationError> ResolveUsageAndSystemIds(Guid usageUuid, Guid systemUuid)
+        {
+            return ResolveUsageId(usageUuid)
+                .Bind(usageId => ResolveSystemId(systemUuid)
+                    .Select(systemId => (usageId, systemId))
+                );
+        }
+
+        private Result<int, OperationError> ResolveUsageId(Guid usageUuid)
```

Comment:
you could make a generic helper for this in stead of two distinct methods

```
private Result<int, OperationError> ResolveId<TEntity>(Guid systemUuid) where TEntity: Entity, IHasUuid
        {
            return _identityResolver.ResolveDbId<TEntity>(systemUuid)
                .Match<Result<int, OperationError>>(id => id, () => new OperationError($"ItSystem with uuid: {typeof(TEntity).Name} was not found", OperationFailure.NotFound));
        }
```
**Also - you could make it an extension method to the service so we can benefit from it elsewhere**